fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsCollectJobFactory.java

```diff
@@ -114,7 +115,16 @@ public static StatisticsCollectJob buildExternalStatisticsCollectJob(String cata
         if (columns == null || columns.isEmpty()) {
             columns = StatisticUtils.getCollectibleColumns(table);
         }
-        return new ExternalFullStatisticsCollectJob(catalogName, db, table, columns,
+
+        List<String> partitionNames;
+        if (!table.isUnPartitioned()) {
+            partitionNames = GlobalStateMgr.getCurrentState().getMetadataMgr().
+                    listPartitionNames(catalogName, db.getFullName(), table.getName());
+        } else {
+            partitionNames = ImmutableList.of(table.getName());
```

Comment:
why use table name as partition name

Reply:
For unpartitioned table, I use table name as partition name